core/collateral/checkpoint.go

```diff
@@ -106,14 +113,31 @@ func (e *Engine) Load(ctx context.Context, data []byte) error {
 			ledgerMovements = append(ledgerMovements, lm)
 			continue
 		}
-		accID := e.accountID(market, balance.Party, balance.Asset, types.AccountTypeGeneral)
-		if _, err := e.GetAccountByID(accID); err != nil {
-			_, _ = e.CreatePartyGeneralAccount(ctx, balance.Party, balance.Asset)
-		}
-		lm, err := e.RestoreCheckpointBalance(
-			ctx, noMarket, balance.Party, balance.Asset, types.AccountTypeGeneral, ub.Clone())
-		if err != nil {
-			return err
+		var (
+			lm  *types.LedgerMovement
+			err error
+		)
+
+		if isVesting {
```

Comment:
can you add a test for this logic please

Reply:
good point, and just at looking at the code I see a bug...

Reply:
that was my way of telling you there was one ;)

Reply:
addressed